### PR TITLE
Make host required field and minor refactor

### LINKED ISSUE #24088: nxos_file_copy copy file localy if no host is set
##### ISSUE TYPE

 - Bug Report
 - Documentation Report

##### COMPONENT NAME

nxos_file_copy

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  config file = /home/adminlocal/noiua-ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.9 (default, Jun 29 2016, 13:08:31) [GCC 4.9.2]
```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### SUMMARY
nxos_file_copy doesn't use the same method to dertermine the remote host to connect to. If no host is given in parameter (which should not be allowed according to the documentation, but it doesn't raise any error), the module connect to the remote host in ssh for the checking/control part but on the host on which it is running for the scp/data part.

##### STEPS TO REPRODUCE
Use a tasks with only local_file set.

```yaml
- nxos_file_copy:
    local_file: "./test_file.txt"
```

##### EXPECTED RESULTS
The module should raise an error as host is a required parameter.
Or, it should use the same host for control and data connection. This would simplify the usage of the module and allow to have the same set of parameters as the others nxos_* module.

##### ACTUAL RESULTS
The file is copied to the localhost and the result is sent. If run again, the file is recopied as the control connection use a connection to the actual device (on which nothing has been copied)

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

##### SUMMARY

Fixes #24088 

##### ISSUE TYPE

 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME

```modules/network/nxos/nxos_file_copy```
##### ANSIBLE VERSION

```
devel 2.4
```

### CODE REVIEW

**Review by ganeshrn — CHANGES_REQUESTED** [review]
I think `host` is a required argument. 
If `host` is not provided the connection attempt
will fail here https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nxos/nxos_file_copy.py#L159-L161

**Comment by trishnaguha** [comment]
@ganeshrn Updated now.

File: lib/ansible/modules/network/nxos/nxos_file_copy.py [hunk 1]
```diff
@@ -100,7 +98,7 @@
     HAS_SCP = False
 
 
-def execute_show_command(command, module, command_type='cli_show'):
+def execute_show_command(command, module):
```
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_file_copy.py]: Any reason why you don't drop this method entirely?

File: lib/ansible/modules/network/nxos/nxos_file_copy.py [hunk 2]
```diff
@@ -189,6 +187,7 @@ def transfer_file(module, dest):
 
 def main():
     argument_spec = dict(
+        host=dict(required=True),
```
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_file_copy.py]: This will get overridden by `argument_spec.update(nxos_argument_spec)` below

**Review by Qalthos — CHANGES_REQUESTED** [review]
I don't think this is the right way to do this. Probably a better way would be through the action plugin, as that's where the values get generated in the first place.

File: lib/ansible/modules/network/nxos/nxos_file_copy.py (hunk 1 above)
Comment (trishnaguha) [reply lib/ansible/modules/network/nxos/nxos_file_copy.py]: Dropping the method would have repetition of code when dealing with `cli` and `nxapi` transport https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nxos/nxos_file_copy.py#L114-L141 .

**Comment by trishnaguha** [comment]
@Qalthos I have updated the PR.

File: lib/ansible/modules/network/nxos/nxos_file_copy.py (hunk 1 above)
Comment (Qalthos) [reply lib/ansible/modules/network/nxos/nxos_file_copy.py]: But regardless of the transport, the only thing it does is `run_commands(module, [command])`. why not replace all `execute_show_command()` with that?

**Comment by trishnaguha** [comment]
Closing the PR and reopening to trigger the shippable again. Looks like there was connection timed out.

**Comment by trishnaguha** [comment]
@Qalthos Right. Removing `execute_show_command` makes sense. Please have a look now.
Thanks!

**Review by Qalthos — APPROVED** [review]

**Pull request merged**